core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -4406,6 +4414,2341 @@ class KafkaApisTest extends Logging {
+    authorizedTopics = authorizedTopics + topicName1
+    authorizedTopics = authorizedTopics + topicName2
+
+    val shareFetchRequestData = new ShareFetchRequestData().
+      setGroupId(groupId).
+      setMemberId(memberId.toString).
+      setShareSessionEpoch(shareSessionEpoch).
+      setTopics(List(
+        new ShareFetchRequestData.FetchTopic().
+          setTopicId(topicId1).
+          setPartitions(List(
+            new ShareFetchRequestData.FetchPartition()
+              .setPartitionIndex(0)
+              .setPartitionMaxBytes(partitionMaxBytes),
+            new ShareFetchRequestData.FetchPartition()
+              .setPartitionIndex(1)
+              .setPartitionMaxBytes(partitionMaxBytes)
+          ).asJava),
+        new ShareFetchRequestData.FetchTopic().
+          setTopicId(topicId2).
+          setPartitions(List(
+            new ShareFetchRequestData.FetchPartition()
+              .setPartitionIndex(0)
+              .setPartitionMaxBytes(partitionMaxBytes),
+          ).asJava),
+      ).asJava)
+
+    val shareFetchRequest = new ShareFetchRequest.Builder(shareFetchRequestData).build(ApiKeys.SHARE_FETCH.latestVersion)
+    val request = buildRequest(shareFetchRequest)
+    // First share fetch request is to establish the share session with the broker.
+    kafkaApis = createKafkaApis(
+      overrideProperties = Map(
+        GroupCoordinatorConfig.NEW_GROUP_COORDINATOR_ENABLE_CONFIG -> "true",
+        ServerConfigs.UNSTABLE_API_VERSIONS_ENABLE_CONFIG -> "true",
+        ShareGroupConfig.SHARE_GROUP_ENABLE_CONFIG -> "true"),
+      raftSupport = true)
+    val fetchResult : mutable.Map[TopicIdPartition, ShareFetchResponseData.PartitionData] =
+      kafkaApis.handleFetchFromShareFetchRequest(
+        request,
+        erroneousAndValidPartitionData,
+        sharePartitionManager,
+        authorizedTopics
+      ).get()
+
+    assertEquals(3, fetchResult.size)
+
+    assertTrue(fetchResult.contains(tp1))
+    val partitionData1 : PartitionData = fetchResult.getOrElse(tp1, null)
+    assertNotNull(partitionData1)
+
+    compareResponsePartitions(
+      0,
+      Errors.NONE.code,
+      Errors.NONE.code,
+      records_t1_p1,
+      expectedAcquiredRecords(0, 9, 1),
+      partitionData1
+    )
+
+    assertTrue(fetchResult.contains(tp2))
+    val partitionData2 : PartitionData = fetchResult.getOrElse(tp2, null)
+    assertNotNull(partitionData2)
+
+    compareResponsePartitionsFetchError(
+      1,
+      Errors.UNKNOWN_TOPIC_OR_PARTITION.code,
+      partitionData2
+    )
+
+    assertTrue(fetchResult.contains(tp3))
+    val partitionData3 : PartitionData = fetchResult.getOrElse(tp3, null)
+    assertNotNull(partitionData3)
+
+    compareResponsePartitionsFetchError(
+      0,
+      Errors.UNKNOWN_TOPIC_OR_PARTITION.code,
+      partitionData3
+    )
+  }
+
+  @Test
+  def testHandleShareFetchFromShareFetchRequestFetchMessagesReturnErrorCode() : Unit = {
```

Comment:
Quite a long name. Could it be sth like `testHandleShareFetchFetchMessagesReturnErrorCode`?